src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrCodeflower.cs

```diff
@@ -375,23 +375,20 @@ private async Task<ForwardFlow> GetLastForwardFlow(string mappingName)
     }
 
     protected async Task<ILocalGitRepo> PrepareRepoAndVmr(
-        SourceMapping mapping,
+        string mappingName,
         string repoRef,
         string vmrRef,
         CancellationToken cancellationToken)
     {
-        var remotes = new[] { mapping.DefaultRef, _sourceManifest.GetRepoVersion(mapping.Name).RemoteUri }
-            .Distinct()
-            .OrderRemotesByLocalPublicOther();
+        await CheckOutVmr(vmrRef);
 
-        ILocalGitRepo repo = await _repositoryCloneManager.PrepareCloneAsync(
-            mapping,
-            [.. remotes],
-            repoRef,
-            cancellationToken);
+        SourceMapping mapping = _dependencyTracker.GetMapping(mappingName);
```

Comment:
we're not using the mapping name for anything else but to get the mapping here, maybe we can keep the original `SourceMapping` parameter and only call `GetMapping` once by calling it before this method. I might be missing some context here tho

Reply:
Yeah - `CheckOutVmr` refreshes the mapping list

Reply:
Ah I see